pkg/operator/pool/controller.go

```diff
@@ -93,20 +93,27 @@ func (c *PoolController) onAdd(obj interface{}) {
 }
 
 func (c *PoolController) onUpdate(oldObj, newObj interface{}) {
-	//oldPool := oldObj.(*Pool)
-	newPool := newObj.(*Pool)
+	oldPool := oldObj.(*Pool)
+	pool := newObj.(*Pool)
+
+	if oldPool.Name != pool.Name {
+		logger.Errorf("failed to update pool %s. name update not allowed", pool.Name)
+		return
+	}
+	if pool.Spec.ErasureCoded.CodingChunks != 0 && pool.Spec.ErasureCoded.DataChunks != 0 {
+		logger.Errorf("failed to update pool %s. erasurecoded update not allowed", pool.Name)
+		return
+	}
```

Comment:
The problem with this is that even though the handler is logging the error and not proceeding,  the pool CRD object will still be updated. The best way is to make use of the CRD validation (alpha in 1.8). https://deploy-preview-5290--kubernetes-io-vnext-staging.netlify.com/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/

This will raise an error once the users create/update the CRD.

Reply:
@kokhang validation seems out of scope for this PR. 1.8 needs to be released first, no?